src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -901,43 +1006,87 @@ public void reduceNeededResources(final Block block)
         }
     }
 
-    private void spawnEntity(@Nullable final Entity entity)
+    /**
+     * Get the entity of an entityInfo object.
+     * @param entityInfo the input.
+     * @return the output object or null.
+     */
+    @Nullable
+    private Entity getEntityFromEntityInfoOrNull(Template.EntityInfo entityInfo)
     {
-        if (entity != null && job instanceof JobBuilder)
+        try
         {
-            final BlockPos pos = ((JobBuilder) job).getStructure().getOffsetPosition();
-
-            if (entity instanceof EntityHanging)
-            {
-                @NotNull final EntityHanging entityHanging = (EntityHanging) entity;
+            return EntityList.createEntityFromNBT(entityInfo.entityData, world);
+        }
+        catch (Exception e)
+        {
+            Log.getLogger().info("Couldn't restore entitiy", e);
+            return null;
+        }
+    }
 
-                entityHanging.posX += pos.getX();
-                entityHanging.posY += pos.getY();
-                entityHanging.posZ += pos.getZ();
-                //also sets position based on tile
-                entityHanging.setPosition(
-                        entityHanging.getHangingPosition().getX(),
-                        entityHanging.getHangingPosition().getY(),
-                        entityHanging.getHangingPosition().getZ());
+    private Boolean spawnEntity(@NotNull final Structure.StructureBlock currentBlock)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "spawnEntity" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)